src/Umbraco.Tests/ObjectExtensionsTests.cs

```diff
@@ -164,28 +168,166 @@ public virtual void CanConvertObjectToSameObject()
+            Assert.AreEqual(100, conv.Result);
+        }
+
+        [Test]
+        public void ConvertToDecimalTest()
+        {
+            var conv = "100".TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = "100.000".TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = "100,000".TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = "100.001".TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100.001m, conv.Result);
+
+            conv = 100m.TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = 100.000m.TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = 100.001m.TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100.001m, conv.Result);
+
+            conv = 100.TryConvertTo<decimal>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+        }
+
+        [Test]
+        public void ConvertToNullableDecimalTest()
+        {
+            var conv = "100".TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = "100.000".TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = "100,000".TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = "100.001".TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100.001m, conv.Result);
+
+            conv = 100m.TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = 100.000m.TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+
+            conv = 100.001m.TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100.001m, conv.Result);
+
+            conv = 100.TryConvertTo<decimal?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(100m, conv.Result);
+        }
+
+        [Test]
+        public void ConvertToDateTimeTest()
+        {
+            var conv = "2016-06-07".TryConvertTo<DateTime?>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(new DateTime(2016, 6, 7), conv.Result);
+        }
+
+        [Test]
+        public void ConvertToNullableDateTimeTest()
+        {
+            var conv = "2016-06-07".TryConvertTo<DateTime>();
+            Assert.IsTrue(conv);
+            Assert.AreEqual(new DateTime(2016, 6, 7), conv.Result);
+        }

```

Comment:
Shouldn't these two test methods of TryConvertTo `DateTime`  / `DateTime?` be opposite?

Reply:
Sure, the naming is a bit off, but at least both cases are covered. :-) I'll update.